drivers/include/mpu9150.h

```diff
@@ -0,0 +1,340 @@
+#define MPU9150_COMP_SELF_TEST      (0x08)
+#define MPU9150_COMP_FUSE_ROM       (0x0F)
+#define MPU9150_COMP_WHOAMI_ANSWER  (0x48)
+/** @} */
+
+/**
+ * @brief Power enum values
+ */
+typedef enum {
+    PWR_OFF = 0x00,
+    PWR_ON = 0x01,
+} mpu9150_pwr_t;
+
+/**
+ * @brief Possible MPU-9150 hardware addresses (wiring specific)
+ */
+typedef enum {
+    HW_ADDR_HEX_68 = 0x68,
+    HW_ADDR_HEX_69 = 0x69,
+} mpu9150_hw_addr_t;
+
+/**
+ * @brief Possible compass addresses (wiring specific)
+ */
+typedef enum {
+    COMP_ADDR_HEX_0C = 0x0C,
+    COMP_ADDR_HEX_0D = 0x0D,
+    COMP_ADDR_HEX_0E = 0x0E,
+    COMP_ADDR_HEX_0F = 0x0F,
+} mpu9150_comp_addr_t;
+
+/**
+ * @brief Possible full scale ranges for the gyroscope
+ */
+typedef enum {
+    GYRO_FSR_250DPS = 0x00,
+    GYRO_FSR_500DPS = 0x01,
+    GYRO_FSR_1000DPS = 0x02,
+    GYRO_FSR_2000DPS = 0x03,
+} mpu9150_gyro_ranges_t;
+
+/**
+ * @brief Possible full scale ranges for the accelerometer
+ */
+typedef enum {
+    ACCEL_FSR_2G = 0x00,
+    ACCEL_FSR_4G = 0x01,
+    ACCEL_FSR_8G = 0x02,
+    ACCEL_FSR_16G = 0x03,
+} mpu9150_accel_ranges_t;
+
+/**
+ * @brief Possible low pass filter values
+ */
+typedef enum {
+    FILTER_188HZ = 0x01,
+    FILTER_98HZ = 0x02,
+    FILTER_42HZ = 0x03,
+    FILTER_20HZ = 0x04,
+    FILTER_10HZ = 0x05,
+    FILTER_5HZ = 0x06,
+} mpu9150_lpf_t;
+
+/**
+ * @brief MPU-9150 result vector struct
+ */
+typedef struct {
+    int16_t x_axis;             /**< X-Axis measurement result */
+    int16_t y_axis;             /**< Y-Axis measurement result */
+    int16_t z_axis;             /**< Z-Axis measurement result */
+} mpu9150_results_t;
+
+/**
+ * @brief Configuration struct for the MPU-9150 sensor
+ */
+typedef struct {
+    mpu9150_pwr_t accel_pwr;            /**< Accel power status (on/off) */
+    mpu9150_pwr_t gyro_pwr;             /**< Gyro power status (on/off) */
+    mpu9150_pwr_t compass_pwr;          /**< Compass power status (on/off) */
+    mpu9150_gyro_ranges_t gyro_fsr;     /**< Configured gyro full-scale range */
+    mpu9150_accel_ranges_t accel_fsr;   /**< Configured accel full-scale range */
+    uint16_t sample_rate;               /**< Configured sample rate for accel and gyro */
+    uint8_t compass_sample_rate;        /**< Configured compass sample rate */
+    uint8_t compass_x_adj;              /**< Compass X-Axis sensitivity adjustment value */
+    uint8_t compass_y_adj;              /**< Compass Y-Axis sensitivity adjustment value */
+    uint8_t compass_z_adj;              /**< Compass Z-Axis sensitivity adjustment value */
+} mpu9150_status_t;
+
+/**
+ * @brief Device descriptor for the MPU-9150 sensor
+ */
+typedef struct {
+    i2c_t i2c_dev;              /**< I2C device which is used */
+    uint8_t hw_addr;            /**< Hardware address of the MPU-9150 */
+    uint8_t comp_addr;          /**< Address of the MPU-9150s compass */
+    mpu9150_status_t conf;      /**< Device configuration */
+} mpu9150_t;
```

Comment:
I'll keep it separated because I think it's a cleaner approach (especially if the configurable features get extended).